### ISSUE TITLE
🐛 Bug Report: [1.33.0] Can't login, auth error: Origin '...' is not allowed

### ISSUE THREAD

**Issue by stephenglass** [backstage/backstage#27729]
### 📜 Description

Upgraded to Backstage v1.33.0, ran a build and deployed it to our Dev environment K8s. When signing in via GitHub auth, I get the following error:
```
Origin 'https://devhub....our FQDN here' is not allowed
```
Reverted back to v1.32.0 and no longer observing the issue.

### 👍 Expected behavior

Login successfully using GitHub Auth.

### 👎 Actual Behavior with Screenshots

Received error message in the UI:
```
Origin '...' is not allowed
```

### 👟 Reproduction steps

Our build process uses `yarn build:backend` to build a bundled app and run in Docker container. The Dockerfile executes a `start-backend.sh` script that ultimately runs a command like this: `exec node packages/backend --config app-config.yaml --config app-config.development.yaml`

### 📃 Provide the context for the Bug.

I can see in the app startup logs its loading the `app-config.yaml` and `app-config.development.yaml` config files. The `app-config.development.yaml` defines the `baseUrl` for the app.

### 🖥️ Your Environment

Backstage v1.33.0
Bundled app running in K8s

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by gaelgoth** [comment]
Similar to this issue: https://github.com/backstage/backstage/issues/27733

**Comment by stephenglass** [comment]
Fixed with #27738
